src/librustc_middle/ty/mod.rs

```diff
@@ -2342,18 +2342,22 @@ impl<'tcx> AdtDef {
         let mut explicit_index = variant_index.as_u32();
```

Comment:
Same here, calling this function on a zero variant enum is a bug, you can just `assert!(!self.variants.is_empty());` at the start of the function to make that explicit.

Reply:
I've changed this to an assert as requested, but I've modified one of the calling functions to check that the variant is within range and return `None`.